docker_publisher.py

```diff
@@ -0,0 +1,547 @@
+        """Initialize a DockerImagePublisherGit with:
+        @git_path: Path to the local git repo clone
+        @git_branch: Branch of the image
+        @path: Path to the directory within the branch which will contain a
+        directory for each architecture, containing:
+        - Dockerfile (including version in a comment)
+        - *.tar.xz: Images"""
+        self.git_path = git_path
+        self.git_branch = git_branch
+        self.path = path
+        self.updated_images = {}
+
+        self.git_call = ["git", "-C", self.git_path]
+
+        ret = subprocess.call(self.git_call + ["fetch", "origin"])
+        if ret != 0:
+            raise DockerFetchException("Could not fetch from origin")
+
+    def releasedDockerImageVersion(self, arch):
+        # Read from git using cat-file to avoid expensive checkout.
+        args = self.git_call + ["cat-file", "--textconv", "origin/%s:%s/%s/Dockerfile" % (self.git_branch, self.path, arch)]
+        with subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as p:
+            version_regex = re.compile("^# Version: (.+)$")
+            for line in p.stdout:
+                match = version_regex.match(line.decode('utf-8').strip())
+                if match:
+                    return match.group(1)
+
+        return "0"
+
+    def generateDockerFile(self, version, filename):
+        """Return the contents of a docker file for the image"""
+        return """FROM scratch
+MAINTAINER Fabian Vogt <fvogt@suse.com>
+# Version: %s
+ADD %s /
+""" % (version, filename)
+
+    def prepareReleasing(self):
+        # Try to delete the old branch
+        try:
+            subprocess.call(self.git_call + ["checkout", "-q", "origin/%s" % (self.git_branch)])
+            subprocess.check_output(self.git_call + ["branch", "-q", "-D", self.git_branch],
+                                    stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError:
+            pass
+
+        # We do an orphan checkout to not carry the history around
+        ret = subprocess.call(self.git_call + ["checkout", "-q", "--orphan", self.git_branch])
+        if ret != 0:
+            raise DockerPublishException("Could not checkout git branch")
+
+        return True
+
+    def addImage(self, version, arch, image_path):
+        target_dir = "%s/%s/%s" % (self.git_path, self.path, arch)
+
+        # Remove all old files
+        for file in glob.glob(target_dir + "/*"):
+            os.remove(file)
+
+        # Re-compress it into the correct location and format
+        targetfilename = "openSUSE-Tumbleweed-%s-%s.tar.xz" % (arch, version)
+
+        # Parse the manifest to get the name of the root tar.xz
+        manifest = json.loads(open(image_path + '/manifest.json').read())
+        layers = manifest[0]['Layers']
+        if len(layers) != 1:
+            raise DockerPublishException("Unexpected count of layers in the image")
+
+        image_layer_file = image_path + '/' + layers[0]
```

Comment:
nit: mixing and matching `'` vs `"` in same function without any visible reason. Not a big deal, but consistency is nice.

Reply:
Indeed, fixed.